### DISCUSSION TITLE
Extend Nuxt3 Global Middleware

### DISCUSSION THREAD

**Discussion by waleedtariq109** [nuxt/nuxt#21320] [Ideas]
Hi everyone,

I was wondering if we could add the layout name as a prefix to the middleware name, similar to how we add global middleware for all pages.

For example:

If we have two layouts, `master.vue` and `default.vue`, and we only want to run the `auth.js` middleware for pages that are using the master layout, we can name the file `auth.master.js`. Now the auth middleware should only run for pages that are using the master layout. Nuxt should be smart enough to detect if the prefix is global, and if so, run that middleware for every page. If the prefix is not global, Nuxt should check if there is any layout that matches the middleware prefix, and if yes, run that middleware for every page that uses that layout.

Alternatively,

Nuxt could provide some sort of composable that allows us to add middleware to a layout.

**Comment by maou-shonen** [comment]
I think it would be better to use something similar to `definePageMeta`.
```ts
defineLayoutMeta({
  middleware: ["some middleware"]
})
```

**Reply by waleedtariq109** [reply]
You are right! It would be great if nuxt provide compiler macro or composable for adding middleware in layouts

**Reply by TheAlexLichter** [reply]
@waleedtariq109 Feel free to submit a feature request ;)

**Reply by waleedtariq109** [reply]
@manniL The feature request is open is currently has enhancement and discussion label on it.
